### PR TITLE
Accept schema and ui schema as inputs

### PR DESCRIPTION
**Pull request by AlexandraBuzila**
Schema and UI schema should be passed as inputs and not via a function call.

Added an example use case to the `index.html`

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
   }
 
-  setJsonFormsInput(
-    schema: JsonSchema,
-    uiSchema?: UISchemaElement,
-    data: any = {}
-  ) {
-    this.jsonformsService.updateCore(Actions.init(data, schema, uiSchema));
+  ngOnChanges(changes: SimpleChanges): void {
+    const schemaObject = JSON.parse(this.schema);
```
Comment (eneufeld) [inline src/app/app.component.ts:20]: can't it happen that you don't have a schema defined?

Comment (AlexandraBuzila) [reply src/app/app.component.ts:20]: It would be a valid use case for JSONForms only if the data is specified, right?

Comment (AlexandraBuzila) [reply src/app/app.component.ts:20]: I can add a `data` input and account for an undefined schema.

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,24 +1,39 @@
...
+  ) {}
 
-  setJsonFormsInput(
-    schema: JsonSchema,
-    uiSchema?: UISchemaElement,
-    data: any = {}
-  ) {
-    this.jsonformsService.updateCore(Actions.init(data, schema, uiSchema));
+  ngOnChanges(changes: SimpleChanges): void {
+    const dataObject = this.data ? JSON.parse(this.data) : undefined;
```
Comment (eneufeld) [inline src/app/app.component.ts:29]: I think we should default data to `{}`
```suggestion
    const dataObject = this.data ? JSON.parse(this.data) : {};
```

Comment (AlexandraBuzila) [reply src/app/app.component.ts:29]: This is not necessary anymore with the latest version of jsonforms :smile:

Comment (AlexandraBuzila) [reply src/app/app.component.ts:29]: ... but the project still uses the old version. Pushing a commit to update the version to 2.4.0

**Pull request merged**